src/Chapter3.hs

```diff
@@ -971,6 +1056,25 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Day = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday deriving (Enum)
+
+isWeekend :: Day -> Bool
+isWeekend Sunday   = True
+isWeekend Saturday = True
+isWeekend _        = False
+
+-- maybe there's a better way to make `succ` wrap here??
+nextDay :: Day -> Day
+nextDay Sunday = Monday
```

Comment:
That is right!
You could also use the instance of `Bounded` in here:

```suggestion
    | day == maxBound = minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂

Reply:
Thanks so much for the comment - I din't know about `Bounded` instances! 🎉